frontend/src/metabase/nav/components/WhatsNewNotification/utils.ts

```diff
@@ -24,7 +24,13 @@ export const getLatestEligibleReleaseNotes = ({
 
   const versions = [versionInfo?.latest]
     .concat(versionInfo?.older)
-    .filter(isNotFalsy);
+    .filter(isNotFalsy)
+    .filter(
+      (
+        record: VersionInfoRecord,
+      ): record is VersionInfoRecord &
+        Required<Pick<VersionInfoRecord, "version">> => Boolean(record.version),
```

Comment:
The type check is incorrect.

```suggestion
        Required<Pick<VersionInfoRecord, "version">> => typeof record.version === 'string',
```